[BE] Pagination for all lists

MihaiDruta opened the pull request:
[BE] Pagination for all lists
https://trello.com/c/0bMghHyt/61-be-pagination-for-all-lists

Implemented pagination for all lists in project.

Commit:
[BE] Pagination for all lists
https://trello.com/c/0bMghHyt/61-be-pagination-for-all-lists

Implemented pagination for all lists in project.

SpartakusMd commented on src/Handlers/ActivityHandler.php:
```diff
@@ -0,0 +1,57 @@
+            $context
+        );
+
+        return new JsonResponse([
```
Return only the array. JsonResponse should be returned from the Controller.

SpartakusMd commented on src/Controller/ActivityController.php:
```diff
@@ -92,26 +115,16 @@ public function __construct(
     /**
      * Get details about and Activity
-     * @Rest\Get("/{id}")
+     * @Rest\Get("/{id}", requirements={"id"="\d+"})
```
Either `{page<\d+>}` or `requirements={"id"="\d+"}` should be used everywhere. Please change it everywhere you will find this.

SpartakusMd commented on src/Handlers/ActivityHandler.php:
```diff
@@ -0,0 +1,57 @@
+        $context = SerializationContext::create()->setGroups(array('ActivityList'));
+
+        $json = $this->serializer->serialize(
+            $paginatedResults,
```
Use `$paginatedResults->getResult()`and remove `$paginatedResults = $paginatedResults->getResult();` from line 39.

SpartakusMd commented on src/Handlers/ActivityUserHandler.php:
```diff
@@ -0,0 +1,57 @@
+        $this->activityUserRepository = $activityUserRepository;
+    }
+
+    public function getApplicantsPaginated(Activity $activity, int $page): JsonResponse
```
Same comments from ActivityHandler.

SpartakusMd commented on src/Handlers/ActivityHandler.php:
```diff
@@ -0,0 +1,57 @@
+        return new JsonResponse([
+            'results' => json_decode($json, true),
+            'numResults' => $numResults,
+            'perPage' => Activity::NUM_ITEMS_PER_PAGE,
```
Move the constant to this class.

SpartakusMd commented on src/Repository/UserRepository.php:
```diff
@@ -47,4 +50,18 @@ public function delete(User $user): void
         $em->flush();
     }
+
+    public function getApplicantsForActivity(Activity $activity): QueryBuilder
```
There is already a method for this: `ActivityUserRepository::getApplicantsForActivity()`

MihaiDruta replied:
The method from ActivityUserRepository didn't work with paginator. So I deleted it and created a new one in UserRepository.

MihaiDruta replied:
https://github.com/doctrine/orm/issues/2596

SpartakusMd replied:
ok

Commit:
[BE] Pagination for all lists
https://trello.com/c/0bMghHyt/61-be-pagination-for-all-lists

Implemented pagination for all lists in project.

SpartakusMd commented on src/Controller/ActivityController.php:
```diff
@@ -670,7 +675,7 @@ public function inviteUserToActivity(
     /**
      * Get a list of all applicants.
-     * @Rest\Get("/{activityId}/applicants/{page<\d+>}", defaults={"page" = 1})
+     * @Rest\Get("/{activityId}/applicants/{page<\d+>}", defaults={"page" = 1}, requirements={"activityId"="\d+"})
```
Remove `<\d+>`

SpartakusMd commented on src/Handlers/ActivityHandler.php:
```diff
@@ -2,17 +2,16 @@
 
 class ActivityHandler
 {
+    public const NUM_ITEMS_PER_PAGE = 4;
```
Use 10 as a default.

SpartakusMd commented on src/Handlers/ActivityUserHandler.php:
```diff
@@ -3,16 +3,15 @@
 
 class ActivityUserHandler
 {
+    public const NUM_APPLICANTS_PER_PAGE = 5;
```
Use 10 as default.

Commit:
[BE] Pagination for all lists
https://trello.com/c/0bMghHyt/61-be-pagination-for-all-lists

Implemented pagination for all lists in project.
